src/openms/source/ANALYSIS/TOPDOWN/FLASHDeconvAlgorithm.cpp

```diff
@@ -0,0 +1,2109 @@
+    defaults_.setValue("max_charge", 100,
+                       "maximum charge state for MS1 spectra (can be negative for negative mode)");
+
+    defaults_.setValue("min_mz", -1.0, "if set to positive value, minimum m/z to deconvolute.");
+    defaults_.setValue("max_mz", -1.0, "if set to positive value, maximum m/z to deconvolute.");
+    defaults_.setValue("min_rt", -1.0, "if set to positive value, minimum RT to deconvolute.");
+    defaults_.setValue("max_rt", -1.0, "if set to positive value, maximum RT to deconvolute.");
+
+    defaults_.setValue("isolation_window", 5.0, "default isolation window with. If the input mzML file does not contain isolation window width information, this width will be used.");
+    defaults_.addTag("isolation_window", "advanced");
+
+    defaults_.setValue("min_isotope_cosine",
+                       DoubleList{.85, .85},
+                       "cosine threshold between avg. and observed isotope pattern for MS1, 2, ... (e.g., -min_isotope_cosine_ 0.8 0.6 to specify 0.8 and 0.6 for MS1 and MS2, respectively)");
+
+    defaults_.setValue("min_qscore",
+                       .0,
+                       "minimum QScore threshold for precursors for MS2. QScore is the probability that a precursor is identified, learned by a logistic regression.");
+
+    defaults_.setValue("min_peaks",
+                       IntList{3, 3},
+                       "minimum number of peaks of consecutive charge states per MS level."
+                       "(e.g., -min_peaks 4 2 to specify 4 and 2 for MS1 and MS2, respectively). "
+                       "This affects only for peaks of highly charged peaks (>8). "
+                       "The peaks of low charges are detected based on m/z distance between isotopes.");
+
+    defaults_.setValue("max_mass_count",
+                       IntList{-1, -1},
+                       "maximum mass count per spec for MS1, 2, ... (e.g., -max_mass_count_ 100 50 to specify 100 and 50 for MS1 and MS2, respectively. -1 specifies unlimited)");
+
+    defaults_.setValue("min_intensity", 10.0, "intensity threshold");
+    defaults_.setValue("rt_window", 180.0, "RT window for MS1 deconvolution");
+    defaultsToParam_();
+  }
+
+  FLASHDeconvAlgorithm::~FLASHDeconvAlgorithm()
+  {
+    if (log_mz_peaks_.empty())
+    {
+      return;
+    }
+    std::vector<LogMzPeak>().swap(log_mz_peaks_);
+  }
+
+  FLASHDeconvAlgorithm& FLASHDeconvAlgorithm::operator=(const FLASHDeconvAlgorithm& fd)
+  {
+    if (this ==& fd)
+    {
+      return *this;
+    }
+    //...
+    return *this;
+  }
+
+  //Calcualte the nominla mass from double mass. Mutiply 0.999497 reduces the rounding error.
+  int FLASHDeconvAlgorithm::getNominalMass(const double mass)
+  {
+    return (int) (mass * 0.999497 + .5);
+  }
+
+
+  //The main function called from outside. precursor_map_for_FLASHIda is used to read FLASHIda information
+  DeconvolvedSpectrum& FLASHDeconvAlgorithm::getDeconvolvedSpectrum(const MSSpectrum& spec,
```

Comment:
sure that this should just be a ref? Not const?

Reply:
I changed this into const ref